Sundar - Phase 2 Summary Dashboard:  Create a grouped-bar chart named Quantity of Materials Used in Projects frontend

sundarmachani opened the pull request:
# Description

Implemented the “Quantity of Materials Used” analytics card on the dashboard to support comparison of material usage across multiple time periods. This includes visual comparison using both bar and line charts, advanced filtering, custom date range support, and detailed material usage drill-down.

**Implements**: Weekly Project Summary Dashboard - Quantity of Materials Used Card

---

## Related PRs (if any)

This frontend PR may be related to upcoming backend updates for material aggregation and filtering by organization/building if separate data endpoints are introduced.

---

## Main Changes Explained

1. **Created a new component** `QuantityOfMaterialsUsed.jsx`
2. **Implements full material usage comparison** using `Chart.js`
   - Bar chart: current period usage
   - Line chart: previous period usage
   - Custom color logic:  
     - 🔴 increase (red)  
     - 🟢 decrease (green)  
     - ⚪ no change (gray)
3. **Interactive drill-down modal** on bar click:
   - Shows total quantity
   - Top-used project
   - Timeline of usage (date + quantity)
4. **Filtering support**:
   - Multi-select dropdown for materials (with search)
   - Organization dropdown (pre-selected from project view)
   - Date filter dropdown:
     - ALL, This/Last Week/Month/Year
     - Custom Date range picker with swap button
5. **Legend logic**:
   - Dynamically shows top 10 visible materials in scroll range
   - Dot colors reflect usage trend
6. **Responsive Design**:
   - Breaks to stacked layout below 1200px
   - Conditional `menuPlacement` for dropdowns on small screens
7. **Dark Mode support**:
   - Integrated with Redux theme selector
   - All charts, dropdowns, tooltips styled using CSS variables
   - Fixed Chart.js tooltip text color in dark mode
8. **Performance + UX improvements**:
   - Scrollbar appears only on overflow
   - Fixed dropdown z-index issues inside nested containers
   - Smooth modal animation with backdrop

---

## How to Test

1. Check out the current branch
2. Run `npm install && npm start`
3. Clear browser cache and site data
4. Log in as an admin user
5. Go to:
   - `Dashboard → Reports → Total Construction Summary`
   - or `http://localhost:3000/bmdashboard/totalconstructionsummary`
6. Try the following:
   - Select different time periods (ALL, This Week, Last Month, Custom Date, etc.)
   - Click on a bar to open usage modal and verify details
   - Select/deselect materials from the dropdown
   - Resize screen below 1200px to test responsiveness
   - Switch to Dark Mode and confirm proper styling + tooltip visibility

---

## Screenshots or Videos of Changes

<img width="471" alt="Screenshot 2025-03-28 at 1 49 36 PM" src="https://github.com/user-attachments/assets/80a50f4e-18bf-4b11-ba66-7025a4cf2a01" />

> ✅ Figma is outdated — please refer to the videos below:

**Requirement Videos**:
- [Loom #1](https://www.loom.com/share/7f8effa2f1ab4a088ac4ec619dd78b2f?sid=65290e5b-8864-4f50-98f3-b633d3abaa39)  
- [Loom #2](https://www.loom.com/share/ad959b0356f74d30814e76e204d0cb28?sid=8695d7d1-01bc-4535-a98e-8e38420b81db)  
- [Loom #3](https://www.loom.com/share/ea30a4f4aa354f0eaf62ada0150965ee?sid=79539d8a-4cdd-43ee-af1a-5e4c85efb148)  
- [Loom #4](https://www.loom.com/share/4bf84c35c9f04f009f76c9074769fa54?sid=f18e1db7-76fb-40d3-8259-7368c92420e9)
- [Loom #5](https://www.loom.com/share/64de6c5edb03422b8ccbdadc9bf79aae?sid=d30aa2c4-65ad-4965-ab3d-c3db82a0c042)

**Testing Walkthrough**:
- [Dropbox video](https://www.dropbox.com/scl/fi/w09yo5uusjzeztsvkm3zs/quantity_of_materials_used_final_2.mov?rlkey=9nmmep543mak25a2nzaxm9swu&st=mu5dwp4s&dl=0)

---

## Notes

- This card depends on:
  - `itemType.name`
  - `updateRecord.date`, `quantityUsed`
  - `project.name` (used for identifying top-used project)
- Please use data that spans multiple time periods to test all features
- ❗ If no data exists for the current period, chart falls back to line graph for previous period
- Modal currently displays the **project where the material was most used**, with a breakdown by date

angad363 reviewed (approved):
I followed the instructions and navigated to bmdashboard/totalconstructionsummary to test the newly implemented Quantity of Materials Used analytics card. The component is displaying correctly, and the various functionalities, including the bar and line charts, filtering options, and date range selection, are working as expected.

Testing Observations:
- The bar chart and line chart are rendering properly and update dynamically based on the selected filters.
- The Material multi-select dropdown is functioning as expected, allowing selection/deselection of materials.
- The Time period filter (ALL, This Week, Last Month, Custom Date, etc.) is correctly updating the graphs.
- The legend accurately reflects the top 5 materials with appropriate colors and labels.
- The custom date picker resets properly when switching between different time periods.
- The color logic (green for increase, red for decrease, gray for no change) is correctly applied to material usage changes.
- The component is fully responsive, adjusting well across different screen sizes.
- Dark mode support is implemented correctly, with proper visibility of tooltips, dropdowns, and text.

I also reviewed the code structure and found it well-organized and up to standard. No major issues were observed. Everything is working fine as per the functionality demonstrated in the attached video.


https://github.com/user-attachments/assets/8b08507b-28bf-4ea2-b191-f3a1b039d7c1

pdp5 reviewed (commented):
I have reviewed the pull request and verified that all changes are working correctly. The QuantityOfMaterialsUsed.jsx component is successfully integrated, and material usage comparison using Chart.js functions as expected.

Charts & Interactivity: The bar chart (current usage) and line chart (previous usage) display accurate data with correct color logic. Clicking a bar opens a drill-down modal with total quantity, top-used project, and usage timeline. Filters and legends work dynamically.

Responsive & Dark Mode: The layout adapts below 1200px, dropdowns adjust placement on smaller screens, and Dark Mode styling is correctly applied.

Performance & UX: Scrollbars appear only when needed, dropdown z-index issues are fixed, and modals animate smoothly.

However, at below 420px, the Tools and Equipment Tracking, Financials, Global Distribution, and Project Status sections have layout issues where their headings extend beyond their containers.


https://github.com/user-attachments/assets/2265be77-c24a-42da-8d63-bceb72ba6bf5

vamsikrishna1704 reviewed (dismissed):
Hi @sundarmachani,
I have reviewed the PR and ran the application by following the guidelines in the PR. I was able to verify the following functionalities:
1. I am able to multi-select the materials from the drop-down list to view it's consumption over different time frames.
2. I am able to click the bars and dots in the graph to view their detailed usage.
3. The bars in the graph are properly differentiated with color code and I can see same color code for the legend.
4. Graph is scrollable to accommodate the selected materials. 
![TotalConsumptionSummaryGraphs](https://github.com/user-attachments/assets/61906b67-4acb-4931-aea2-29db4a135f64)

sundarmachani commented:
> I have reviewed the pull request and verified that all changes are working correctly. The QuantityOfMaterialsUsed.jsx component is successfully integrated, and material usage comparison using Chart.js functions as expected.
> 
> Charts & Interactivity: The bar chart (current usage) and line chart (previous usage) display accurate data with correct color logic. Clicking a bar opens a drill-down modal with total quantity, top-used project, and usage timeline. Filters and legends work dynamically.
> 
> Responsive & Dark Mode: The layout adapts below 1200px, dropdowns adjust placement on smaller screens, and Dark Mode styling is correctly applied.
> 
> Performance & UX: Scrollbars appear only when needed, dropdown z-index issues are fixed, and modals animate smoothly.
> 
> However, at below 420px, the Tools and Equipment Tracking, Financials, Global Distribution, and Project Status sections have layout issues where their headings extend beyond their containers.
> 
>  PR3313.mp4

Hi @pdp5, The issue you are stating is at the smallest displays or mobile displays, This dashboard isn't compatible for this as this follows the grid approach might need to check and stack them up however the behaviour isn't known.

> I have reviewed the pull request and verified that all changes are working correctly. The QuantityOfMaterialsUsed.jsx component is successfully integrated, and material usage comparison using Chart.js functions as expected.
> 
> Charts & Interactivity: The bar chart (current usage) and line chart (previous usage) display accurate data with correct color logic. Clicking a bar opens a drill-down modal with total quantity, top-used project, and usage timeline. Filters and legends work dynamically.
> 
> Responsive & Dark Mode: The layout adapts below 1200px, dropdowns adjust placement on smaller screens, and Dark Mode styling is correctly applied.
> 
> Performance & UX: Scrollbars appear only when needed, dropdown z-index issues are fixed, and modals animate smoothly.
> 
> However, at below 420px, the Tools and Equipment Tracking, Financials, Global Distribution, and Project Status sections have layout issues where their headings extend beyond their containers.
> 
>  PR3313.mp4

Hi @pdp5, the concern you stated in this is applicable to mobile devices as this Page uses a grid approach and isn't compatible. Might need to stack up every card that is available on this page. For a better experience, I will need to check the design options for stacking up. Thanks

Dharmik1710 commented:
I followed the setup instructions and successfully tested the Quantity of Materials Used grouped-bar chart on the Summary Dashboard. The chart correctly displays current period usage (bar chart) and previous period comparison (line chart), with color-coded trends (🔴 increase, 🟢 decrease, ⚪ no change).

Observations:
Bars and lines display correctly with accurate values.
Clicking a bar opens a smooth interactive modal with detailed material usage breakdown.
Multi-select materials, date filters, and organization selection update the chart dynamically.
UI elements adapt properly, and tooltip text color is correctly fixed.
Checked out the dark mode as well

Setup Notes:
Checked out sundar_grouped_bar_chart_for_the_quantity_of_materials
Installed dependencies & cleared cache
Verified chart updates dynamically across different views

Great implementation! A well-structured, interactive feature that enhances the dashboard experience. 

<img width="1440" alt="PR 3313 - 1" src="https://github.com/user-attachments/assets/b8efcd27-32dc-43b9-ac22-ef30206cb12b" />
<img width="1440" alt="PR 3313 - 2" src="https://github.com/user-attachments/assets/a738658a-381f-4ae1-b728-a8f338d8ba43" />
<img width="1440" alt="PR 3313 - 3" src="https://github.com/user-attachments/assets/4257b822-737b-4f62-8983-7037938847ad" />

sundarmachani commented:
> > I have reviewed the pull request and verified that all changes are working correctly. The QuantityOfMaterialsUsed.jsx component is successfully integrated, and material usage comparison using Chart.js functions as expected.
> > Charts & Interactivity: The bar chart (current usage) and line chart (previous usage) display accurate data with correct color logic. Clicking a bar opens a drill-down modal with total quantity, top-used project, and usage timeline. Filters and legends work dynamically.
> > Responsive & Dark Mode: The layout adapts below 1200px, dropdowns adjust placement on smaller screens, and Dark Mode styling is correctly applied.
> > Performance & UX: Scrollbars appear only when needed, dropdown z-index issues are fixed, and modals animate smoothly.
> > However, at below 420px, the Tools and Equipment Tracking, Financials, Global Distribution, and Project Status sections have layout issues where their headings extend beyond their containers.
> > PR3313.mp4
> 
> Hi @pdp5, the concern you stated in this is applicable to mobile devices as this Page uses a grid approach and isn't compatible. Might need to stack up every card that is available on this page. For a better experience, I will need to check the design options for stacking up. Thanks

Hi, @pdp5 I have fixed the grid issues now the cards will stack up on a smaller screen. Thanks

humemah commented:
I’ve tested the implementation, and everything works as expected. The charts, filtering options, and interactive drill-downs function smoothly. Everything is working as expected
![image](https://github.com/user-attachments/assets/aa8f3a58-5bf1-4b0f-9368-74de20880023)
![image](https://github.com/user-attachments/assets/a3b5632d-b3b9-41aa-a49c-46663b02b48d)
![image](https://github.com/user-attachments/assets/76a1757f-83ae-440f-8301-49d16e0563f7)

Amalesh-A reviewed (dismissed):
I checked out your branch, reviewed your code, followed your instructions, checked multiple criteria, and observed that the functionality was running successfully.
![1](https://github.com/user-attachments/assets/8adbd8ce-5dd3-4194-b34a-cc437d843dad)

karnati007 reviewed (dismissed):
![SundarPR](https://github.com/user-attachments/assets/6ff7a281-48f8-47cc-b012-1fd97bf35cb6)
Checked out the current branch, ran npm install && npm start, and cleared browser cache and site data. Logged in as an admin user and navigated to Dashboard → Reports → Total Construction Summary or http://localhost:3000/bmdashboard/totalconstructionsummary.

Tested time period selection (ALL, This Week, Last Month, Custom Date, etc.), and data updated correctly.

Clicked on a bar in the chart, verified that the usage modal opens and displays the correct details.

Checked material selection/deselection from the dropdown, and it functions as expected.
Switched to Dark Mode, verified styling, and ensured tooltips are visible.
Everything works as expected.

linh2020 reviewed (dismissed):
Tested on: http://localhost:3000/bmdashboard/totalconstructionsummary

Multi-select materials & dynamic chart update.
Date filters (This Week, Last Month, Custom) respond correctly.
Bar click opens drill-down modal with correct data.
Color-coded trend indicators work as intended.
Fully responsive UI + Dark Mode styling verified.
Scrollable legend and smooth animations.

Everything works as expected.
Great work!

![2025-04-03 23_51_19-Total Construction Summary and 1 more page - Your Microsoft Edge - Microsoft​ Ed](https://github.com/user-attachments/assets/422520e0-d7c8-4ac9-9f8a-6a44a363f571)

![2025-04-03 23_51_36-Total Construction Summary and 1 more page - Your Microsoft Edge - Microsoft​ Ed](https://github.com/user-attachments/assets/7918daac-75a5-41a3-b92e-b311881ebe76)

![2025-04-03 23_51_50-Total Construction Summary and 1 more page - Your Microsoft Edge - Microsoft​ Ed](https://github.com/user-attachments/assets/27dc6a3c-0911-4496-bbbd-48ce7bdddd6f)

![2025-04-03 23_52_01-Total Construction Summary and 1 more page - Your Microsoft Edge - Microsoft​ Ed](https://github.com/user-attachments/assets/beedec95-f01e-45f9-b71b-37367831dbd0)

![2025-04-03 23_52_40-Total Construction Summary and 1 more page - Your Microsoft Edge - Microsoft​ Ed](https://github.com/user-attachments/assets/1a9d8149-5c13-4cd9-a377-759c48b9488b)

![2025-04-03 23_53_55-Total Construction Summary and 1 more page - Your Microsoft Edge - Microsoft​ Ed](https://github.com/user-attachments/assets/cb5c9c14-a09d-445c-af05-b3c3e9f318db)

![2025-04-03 23_54_21-Sundar - Phase 2 Summary Dashboard_ Create a grouped-bar chart named Quantity of](https://github.com/user-attachments/assets/a83b751e-555c-4969-9199-4a25e6fa822d)

![2025-04-03 23_54_51-Total Construction Summary and 1 more page - Your Microsoft Edge - Microsoft​ Ed](https://github.com/user-attachments/assets/83899301-aba2-4236-99e0-4f2ae57e4e25)

![2025-04-03 23_55_06-Total Construction Summary and 1 more page - Your Microsoft Edge - Microsoft​ Ed](https://github.com/user-attachments/assets/5925a771-310f-4005-80a3-459cda352652)

JuliaHa0902 reviewed (approved):
I have tested the PR. Everything works great.
 - Custom color logic matches the trend of before and after
 - When I clicked on bar, it shows the modal with the total quantity, top-used project and timeline of usage
 - Filtering works great. I have tried all the filters. Custom date works as well
 - Top 10 visible materials are shown dynamically
 - Dark mode is supported
 - The chart is responsive
 
![Screenshot 2025-04-16 190455](https://github.com/user-attachments/assets/1f651268-c88d-486b-80dd-fbaedc8c069a)
![Screenshot 2025-04-16 190632](https://github.com/user-attachments/assets/a3aa1ff2-b80b-4d97-adb7-0f8e6782a715)
![Screenshot 2025-04-16 190722](https://github.com/user-attachments/assets/d98b4b1e-120b-4bf1-b830-3c5deb730863)

JuliaHa0902 reviewed (commented):
I just have a few tiny comments:
 - The "Top 10 of .." probably should be renamed as "Top 10 Visibility of ...". I was confused at first about how the top 10 keeps changing.
 - The total quantity showed in the modal doesn't match the number of the tooltips when you hover over the material. For example, in here I see that This Year Usage of sand is 1015. 
 
![Screenshot 2025-04-16 191823](https://github.com/user-attachments/assets/5656ba8e-14ee-4b41-ab57-1d21197e1b0a)
However when I click on the bar, the modal shows that total quantity is 100. I don't understand where the number 1015 comes from. 
![Screenshot 2025-04-16 191956](https://github.com/user-attachments/assets/fec2af43-2c94-490e-ab18-7e221d8fb56c)
I also notice that it shows the same information when I clicked on Last Year usage and This Year usage. I don't understand the logic behind so I just want to bring your attention. Is this intended? Maybe we can reword the phrase a little bit so users don't misundertand the number. For example, clarify the period of Total Quantity (Total Quantity From ... To ..., Highest use in ... From ... To....).
 - The tooltip of the chart explanation is too long. In small screen, it got cut the first part of the tooltip. My recommendation is that for a long explanation like this, maybe we should use a modal with scrollable?
 
![Screenshot 2025-04-16 192458](https://github.com/user-attachments/assets/95f55fe2-44c2-40cd-9264-7da5ed266bd9)

sundarmachani commented:
> I just have a few tiny comments:
> 
> * The "Top 10 of .." probably should be renamed as "Top 10 Visibility of ...". I was confused at first about how the top 10 keeps changing.
> * The total quantity showed in the modal doesn't match the number of the tooltips when you hover over the material. For example, in here I see that This Year Usage of sand is 1015.
> 
> ![Screenshot 2025-04-16 191823](https://private-user-images.githubusercontent.com/43044560/434570582-5656ba8e-14ee-4b41-ab57-1d21197e1b0a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ5MzAzMDgsIm5iZiI6MTc0NDkzMDAwOCwicGF0aCI6Ii80MzA0NDU2MC80MzQ1NzA1ODItNTY1NmJhOGUtMTRlZS00YjQxLWFiNTctMWQyMTE5N2UxYjBhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE3VDIyNDY0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRjZThkMzE1ZDc4NzIzOWVhMjM5NTYzMGIyNjA5OGYwMjE5NzQyN2NlOTcwZjZjNTljY2IyNzczNzRjYjcxYzYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.w_c5WlGHb6Z_cu716SCTYb0dCLKoCut5r_4CL1MsvcI) However when I click on the bar, the modal shows that total quantity is 100. I don't understand where the number 1015 comes from. ![Screenshot 2025-04-16 191956](https://private-user-images.githubusercontent.com/43044560/434570752-fec2af43-2c94-490e-ab18-7e221d8fb56c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ5MzAzMDgsIm5iZiI6MTc0NDkzMDAwOCwicGF0aCI6Ii80MzA0NDU2MC80MzQ1NzA3NTItZmVjMmFmNDMtMmM5NC00OTBlLWFiMTgtN2UyMjFkOGZiNTZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE3VDIyNDY0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU0OWZiZjVhMmQ2ZjVlNGZhMWRiMDc0YWQ4OWY2ZmU1ZDU2MGMyNWZiNjg3YWQ2YWJkZGQwN2ZiMjcyZmVkYmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.sohvzoG7PMiXonCwXPb5KluGxohpyk7s1fvWIvcmJKs) I also notice that it shows the same information when I clicked on Last Year usage and This Year usage. I don't understand the logic behind so I just want to bring your attention. Is this intended? Maybe we can reword the phrase a little bit so users don't misundertand the number. For example, clarify the period of Total Quantity (Total Quantity From ... To ..., Highest use in ... From ... To....).
> 
> * The tooltip of the chart explanation is too long. In small screen, it got cut the first part of the tooltip. My recommendation is that for a long explanation like this, maybe we should use a modal with scrollable?
> 
> ![Screenshot 2025-04-16 192458](https://private-user-images.githubusercontent.com/43044560/434571751-95f55fe2-44c2-40cd-9264-7da5ed266bd9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ5MzAzMDgsIm5iZiI6MTc0NDkzMDAwOCwicGF0aCI6Ii80MzA0NDU2MC80MzQ1NzE3NTEtOTVmNTVmZTItNDRjMi00MGNkLTkyNjQtN2RhNWVkMjY2YmQ5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE3VDIyNDY0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBmM2I4YmRlYTQwZjJjODViOTVmZjI1Nzc3M2M0Y2M4MTY0MzcwYWM5OGFkY2MzZTkwMzU0ZDdkZWU1MTVkYTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.G5dsSO384toQHDesAuushwNXqZ_5ueGT4O3Z4-mY7UI)

Hi @JuliaHa0902, I’ve updated the proposed items:

Added a scrollbar to the info ("i") button tooltip — you should now be able to scroll through the text if it overflows.

Updated the legend label on the chart’s button for clarity. It now says: Visible Top X of Xrange (instead of "Top X of Xrange").

The modal pop-up that appears when clicking on a bar now displays the project where the selected material had the highest usage.
For example, Sand1 has a total quantity of 1015, which is distributed across all available projects (in this case, Building 1, 2, and 3).
The usage is calculated based on the date range selected via the dropdown. Only the materials used within that selected range are aggregated, and the project with the highest usage among them is shown in the modal.

JuliaHa0902 reviewed (changes requested):
Hi Sundar, I have reviewed the new updated code. It looks great!
I saw that the top 10 is now labeled as Visible Top 10
![Screenshot 2025-04-17 184322](https://github.com/user-attachments/assets/b086e173-44e2-4d60-9654-90e5b493d993)
The tooltip can be scrollable now
![Screenshot 2025-04-17 184526](https://github.com/user-attachments/assets/9d00cc5e-3ada-4189-aefe-a0bb44849aae)
However I found a new issue:
 - The type of rendered chart seems to be affected by what was selected before. For example, when I switch between Last Month and This Month, the chart sometimes get rendered as 2 line charts. Please see the following video

https://github.com/user-attachments/assets/4d7291c9-6582-4ef7-818d-a272fe3622b6

- The selected options get cut if the name is longer than the bar. Maybe we can just change the text color instead?
![Screenshot 2025-04-17 184133](https://github.com/user-attachments/assets/3733935f-3406-48a0-b5e4-d2877bf690c4)
 - Not an error but more like a suggested feature: I suggest adding an All Materials option, so we can easily switch between selecting a few materials to all materials. If you don't think this is a good idea, please feel free to ignore it :D
 
![Screenshot 2025-04-17 184241](https://github.com/user-attachments/assets/caf367ff-9947-44b7-add2-14387e5298ac)

sundarmachani commented:
> Hi Sundar, I have reviewed the new updated code. It looks great! I saw that the top 10 is now labeled as Visible Top 10 ![Screenshot 2025-04-17 184322](https://private-user-images.githubusercontent.com/43044560/435001181-b086e173-44e2-4d60-9654-90e5b493d993.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ5NTQyNDIsIm5iZiI6MTc0NDk1Mzk0MiwicGF0aCI6Ii80MzA0NDU2MC80MzUwMDExODEtYjA4NmUxNzMtNDRlMi00ZDYwLTk2NTQtOTBlNWI0OTNkOTkzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE4VDA1MjU0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg2Y2NhNmJlOGNjMjg0NDgzZWZjMTJhNGEyMmFmZWIzNTQ4MjdhNzEzNzY5MjcyMzI2NGYxZDI5MzM3NWU0MjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.COoUScTAfJWattrDjZ5lFZDlCaX8Di2cp5qw2tCrIzs) The tooltip can be scrollable now ![Screenshot 2025-04-17 184526](https://private-user-images.githubusercontent.com/43044560/435001232-9d00cc5e-3ada-4189-aefe-a0bb44849aae.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ5NTQyNDIsIm5iZiI6MTc0NDk1Mzk0MiwicGF0aCI6Ii80MzA0NDU2MC80MzUwMDEyMzItOWQwMGNjNWUtM2FkYS00MTg5LWFlZmUtYTBiYjQ0ODQ5YWFlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE4VDA1MjU0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEyMzlkNzkxN2NhY2M3YzRmZTVlYzJlMGYwMzc5YjFiZmM2M2E0MjgwN2M0MWY0Yjc2YTgwMDhhNGMxMmI1M2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.p6lb_vg5mVVTOgXVg6pgZR8JRe8-oNfeTk0lKenNUAs) However I found a new issue:
> 
> * The type of rendered chart seems to be affected by what was selected before. For example, when I switch between Last Month and This Month, the chart sometimes get rendered as 2 line charts. Please see the following video
> 
>  PR.3313.mp4 
> 
> * The selected options get cut if the name is longer than the bar. Maybe we can just change the text color instead?
>   ![Screenshot 2025-04-17 184133](https://private-user-images.githubusercontent.com/43044560/435001823-3733935f-3406-48a0-b5e4-d2877bf690c4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ5NTQyNDIsIm5iZiI6MTc0NDk1Mzk0MiwicGF0aCI6Ii80MzA0NDU2MC80MzUwMDE4MjMtMzczMzkzNWYtMzQwNi00OGEwLWI1ZTQtZDI4NzdiZjY5MGM0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE4VDA1MjU0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdkNDIzYjRlNjEyYjZmNWIyNGNmODU0NDUzMzZiNGIxNTVmNjc0MjYyYjdlMzhiZjU3NGI2MzI1ODM1ZTk3NDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Tm1HZovQOr46S0qxjQPtYuByRNXT3zZMNpwWZQuK27E)
> * Not an error but more like a suggested feature: I suggest adding an All Materials option, so we can easily switch between selecting a few materials to all materials. If you don't think this is a good idea, please feel free to ignore it :D
> 
> ![Screenshot 2025-04-17 184241](https://private-user-images.githubusercontent.com/43044560/435002071-caf367ff-9947-44b7-add2-14387e5298ac.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQ5NTQyNDIsIm5iZiI6MTc0NDk1Mzk0MiwicGF0aCI6Ii80MzA0NDU2MC80MzUwMDIwNzEtY2FmMzY3ZmYtOTk0Ny00NGI3LWFkZDItMTQzODdlNTI5OGFjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE4VDA1MjU0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEzMGU3MjU5OGI2NzEzN2U5YmZhYWY1NTNlNTUwNGRhNDI5ZTAxYjM4M2M2MTdiNjkzMmM3NmViMjZlNTZkNjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.aq8Cdfh_U9W0nckl4udVcA6CLa9XMtdG1H8q1sSK-CE)

Hello @JuliaHa0902 
I improved the speed to fix the rendering issue in the charts, and also fixed the partial menu item when hovered. 
Thank you.

JuliaHa0902 reviewed (approved):
Everything looks great! The chart renders so fast now and the problem about the chart type is gone.
![Screenshot 2025-04-18 084718](https://github.com/user-attachments/assets/b7e79e81-e60f-45f3-b607-b3f83b16c7fa)
![Screenshot 2025-04-18 084730](https://github.com/user-attachments/assets/4ee93815-17cd-4cd9-a22f-77ffc8c9cd0e)
The selected bar now cover the whole text
![Screenshot 2025-04-18 084740](https://github.com/user-attachments/assets/0c193413-423a-48e3-ad4c-f8f596ad24cd)
All other functionalities are working as intended. Great job! I approved :D

one-community commented:
Thank you all, merging!
